typespec-tests/src/main/java/com/cadl/patch/models/Salmon.java

```diff
@@ -294,10 +279,10 @@ public static Salmon fromJson(JsonReader jsonReader) throws IOException {
                 }
             }
             Salmon deserializedSalmon = new Salmon();
-            deserializedSalmon.setId(id);
-            deserializedSalmon.setName(name);
-            deserializedSalmon.setAge(age);
-            deserializedSalmon.setColor(color);
+            deserializedSalmon.id = id;
```

Comment:
Something seems off here as the model doesn't have any constructor parameters but is using the constructor style deserialization with having variables for all the deserialized values. Won't fix in this PR as the behavior hasn't changed but this can get cleaned up.